thriftpool/doc_test.go

```diff
@@ -0,0 +1,125 @@
+package thriftpool_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/thriftpool"
+)
+
+// In real code these should be coming from either config file or flags instead.
+const (
+	remoteAddr    = "host:port"
+	socketTimeout = time.Millisecond * 10
+
+	minConnections = 50
+	maxConnections = 100
+
+	clientTTL = time.Minute * 5
+
+	poolGaugeInterval = time.Second * 10
+)
+
+// BEGIN THRIFT GENERATED CODE SECTION
+//
+// In real code this section should be from thrift generated code instead,
+// but for this example we just define some placeholders here.
+
+type MyEndpointRequest struct{}
+
+type MyEndpointResponse struct{}
+
+type MyService interface {
+	MyEndpoint(ctx context.Context, req *MyEndpointRequest) (*MyEndpointResponse, error)
+}
+
+func NewMyServiceClientFactory(_ thrift.TTransport, _ thrift.TProtocolFactory) MyService {
+	// In real code this certainly won't return nil.
+	return nil
+}
+
+// END THRIFT GENERATED CODE SECTION
+
+type Client interface {
+	thriftpool.Client
+
+	MyService
+}
+
+type clientImpl struct {
+	MyService
+
+	*thriftpool.TTLClient
+}
+
+func newClient(addr string) (Client, error) {
+	trans, err := thrift.NewTSocketTimeout(addr, socketTimeout)
+	if err != nil {
+		return nil, err
+	}
+	err = trans.Open()
+	if err != nil {
+		return nil, err
+	}
+	protoFactory := thrift.NewTHeaderProtocolFactory()
+	client := NewMyServiceClientFactory(trans, protoFactory)
```

Comment:
Would you consider using:

```suggestion
	client := NewMyServiceClient(thriftclient.NewMonitoredClientFromFactory(trans, protoFactory))
```

so we set other devs up to use that by default?

Reply:
That change was not merged in yet 😅

Reply:
:badpokerface:

_yet_

https://github.com/reddit/baseplate.go/commit/be09f33afa2edea73bbec51bb096648796594450

Reply:
:shakes-fist:

Reply:
And the line is soooooo long!